tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -833,13 +833,37 @@ func testCheckCache(t *testing.T, visit func() error, app string) {
 }
 
 // nolint: unparam
-func fetchRequestCount(t *testing.T, promAPI v1.API, service string) (prior429s float64, prior200s float64, value model.Value) {
+func fetchRequestCount(t *testing.T, promAPI v1.API, service string, totalReqExpected float64) (prior429s float64, prior200s float64, value model.Value) {
 	var err error
 	t.Log("Establishing metrics baseline for test...")
-	query := fmt.Sprintf("istio_requests_total{%s=\"%s\"}", destLabel, fqdn(service))
-	t.Logf("prometheus query: %s", query)
-	value, err = promAPI.Query(context.Background(), query, time.Now())
-	if err != nil {
+
+	retry := util.Retrier{
+		BaseDelay: 30 * time.Second,
+		Retries:   4,
+	}
+
+	retryFn := func(_ context.Context, i int) error {
+		t.Helper()
+		t.Logf("Trying to find metrics via promql (attempt %d)...", i)
+		query := fmt.Sprintf("istio_requests_total{%s=\"%s\"}", destLabel, fqdn(service))
```

Comment:
should we just `sum()` with an additional dimension of `reporter="destination"` in the query (instead of relying on the wonky `vectorValue`)?

Reply:
Thanks..Fixed